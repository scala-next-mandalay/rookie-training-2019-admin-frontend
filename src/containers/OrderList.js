import { connect } from 'react-redux';
import OrderList from '../components/OrderList';
import {fetchAllOrders , setSearchText} from '../modules/orders';
import { fetchAllOrderItems } from '../modules/orderitems';

// const _filter = (rows, searchText) => {
//   console.log("searchtext:",searchText);
//   if (!searchText) {
//     return  rows;
//   }
//   else 
//   {
//     return searchText;
//   }
  
//   // const filteredRows = [];
//   // for (const n of rows) {
//   //   //filter by searchText.
//   //   if (searchText) {
//   //     if (n.first_name.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.last_name.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || (n.first_name+" "+n.last_name).toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || (n.first_name+n.last_name).toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.address1.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.address2.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.country.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.state.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //       || n.city.toLowerCase().indexOf(searchText.toLowerCase()) > -1
//   //     ) {
//   //       filteredRows.push(n);
//   //     }
//   //   }
//   // }
//   // return filteredRows;
// };


export default connect(
  (state) => ({
    orders: state.orders.rows,
    searchText: state.orders.searchText,
  }),
  (dispatch) => ({
    setSearchText: (text) =>  dispatch(setSearchText(text)),
    fetchAllOrders: () => dispatch(fetchAllOrders()),
    fetchAllOrderItems: (id) => dispatch(fetchAllOrderItems(id)),
  })
)(OrderList);