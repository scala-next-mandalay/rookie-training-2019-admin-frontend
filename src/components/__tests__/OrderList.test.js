/*global jest,expect,MouseEvent,Event*/
import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import OrderList from '../OrderList';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});

let num = 0 ;let newOrders = [];
const fetchAllOrders = jest.fn((val) => {
    num = val;
    //console.log('Income Value :',num)
    if(num >= 10)
    {
      newOrders = [{address1: "11"}];
    }
    else 
      newOrders = [{address1: "1"},
                  {address1: "2"},
                  {address1: "3"},
                  {address1: "4"},
                  {address1: "5"},
                  {address1: "6"},
                  {address1: "7"},
                  {address1: "8"},
                  {address1: "9"},
                  {address1: "10"},
                  {address1: "11"}];
});
    
const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;

    describe("OrderList component", () => {
      let orders = [{address1: "98 street", address2: "Min Buu", city: "Magway" , country: "Myanmar",
                    first_name: "Thet Oo", id: 1 , last_name: "Maung" ,state: "Magway" ,total_price: 6000 },
                  {address1: "Bauk taw", address2: "POL", city: "Mandalay" , country: "Myanmar",
                    first_name: "Soe", id: 2 , last_name: "Thu" ,state: "Mandalay" ,total_price: 2000 }];
      
      it('matches the snapshot', () => {
        const OrderListSnapshot = renderer.create(
          <Parent>
            <OrderList
              orders={orders}
              fetchAllOrders={fetchAllOrders}
            />
          </Parent>
        ).toJSON();
        expect(OrderListSnapshot).toMatchSnapshot();
      });
      
    });
    
    describe("Order Searching", () => {
      it('should have 1 textfield , 3 buttons and 5 links', () => {
        let orders = [{address1: "98 street", address2: "Min Buu", city: "Magway" , country: "Myanmar",
                    first_name: "Thet Oo", id: 1 , last_name: "Maung" ,state: "Magway" ,total_price: 6000 },
                  {address1: "Bauk taw", address2: "POL", city: "Mandalay" , country: "Myanmar",
                    first_name: "Soe", id: 2 , last_name: "Thu" ,state: "Mandalay" ,total_price: 2000 }];
        
        act(() => {
          ReactDOM.render((
            <Parent>
                <OrderList 
                  orders={orders}
                  fetchAllOrders={fetchAllOrders}
                />
            </Parent>
          ), container);
        });
        
        const inputArr = document.querySelectorAll('input');
        expect(inputArr).toHaveLength(1);
        expect(inputArr[0].id).toBe("standard-search");
        const button = container.querySelectorAll('Button');
        expect(button[1].textContent.toUpperCase()).toBe("PREV");
        expect(button[2].textContent.toUpperCase()).toBe("NEXT");
        const link = document.querySelectorAll('a');
        expect(link).toHaveLength(5);
      });
      
      it('check working searching', () => {
        let orders = [{address1: "98 street", address2: "Min Buu", city: "Magway" , country: "Myanmar",
                    first_name: "Thet Oo", id: 1 , last_name: "Maung" ,state: "Magway" ,total_price: 6000 },
                  {address1: "Bauk taw", address2: "POL", city: "Mandalay" , country: "Myanmar",
                    first_name: "Soe", id: 2 , last_name: "Thu" ,state: "Mandalay" ,total_price: 2000 }];

        const setSearchText = jest.fn((setText) => {
          console.log('SearchText :',setText)
          for (const n of orders) {
            if (n.address1.toLowerCase().indexOf(setText.toLowerCase())) 
            {
                console.log('Hoo :',setText)
            }
          }
          return orders;
        });
        
        act(() => {
          ReactDOM.render((
            <Parent>
                <OrderList 
                  orders={orders}
                  fetchAllOrders={fetchAllOrders}
                  setSearchText={setSearchText}
                />
            </Parent>
          ), container);
        });
        
        const inputArr = document.querySelectorAll('input');
        console.log('Before :',inputArr[0].value);
        inputArr[0].value = "Min";
        act(() => {
          inputArr[0].dispatchEvent(new MouseEvent('change', { bubbles: true }));
        });
        console.log('After :',inputArr[0].value)
        expect(inputArr[0].value).toBe("Min");
        const searchButton = container.querySelectorAll('Button');
        act(() => {
          searchButton[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
        });
        
        const expectedOrders = [
          {address1: "98 street", address2: "Min Buu", city: "Magway" , country: "Myanmar",
                    first_name: "Thet Oo", id: 1 , last_name: "Maung" ,state: "Magway" ,total_price: 6000 }
        ];
        
        expect(setSearchText).toHaveBeenCalledWith();
        expect(orders).toStrictEqual(expectedOrders);
      });
      
      // it('testing Prev and Next Button', () => {
      //   let orders = [{address1: "1"},
      //             {address1: "2"},
      //             {address1: "3"},
      //             {address1: "4"},
      //             {address1: "5"},
      //             {address1: "6"},
      //             {address1: "7"},
      //             {address1: "8"},
      //             {address1: "9"},
      //             {address1: "10"},
      //             {address1: "11"}];
        
      //   act(() => {
      //     ReactDOM.render((
      //       <Parent>
      //           <OrderList 
      //             orders={orders}
      //             fetchAllOrders={fetchAllOrders}
      //           />
      //       </Parent>
      //     ), container);
      //   });
      //   const Button = container.querySelectorAll('Button');
      //   act(() => {
      //     Button[2].dispatchEvent(new MouseEvent('click', { bubbles: true }));
      //   });
      //   expect(fetchAllOrders).toHaveBeenCalled();
      //   console.log('NextOrders :',newOrders)
      //   console.log('Next Val :',num)
      //   console.log('Prev ? :',Button[1].textContent)
      //   act(() => {
      //     Button[1].dispatchEvent(new MouseEvent('click', { bubbles: true }));
      //   });
      //   expect(fetchAllOrders).toHaveBeenCalled();
      //   expect(fetchAllOrders).toHaveBeenCalled();
      //   console.log('PrevOrders :',newOrders)
      //   console.log('Prev Val :',num)
      // });
    });