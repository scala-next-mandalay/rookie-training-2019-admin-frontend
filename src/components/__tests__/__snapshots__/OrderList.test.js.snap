// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList component matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="makeStyles-title-134"
        >
          <div
            className="input-group"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={true}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  disabled={false}
                  id="standard-search"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search"
                  required={false}
                  type="search"
                  value=""
                />
              </div>
            </div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              width="50"
            >
              <span
                className="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          className="makeStyles-spacer-135"
        />
        <div>
          <div
            className="MuiBox-root MuiBox-root-205"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              tabIndex={-1}
              type="button"
              width="50"
            >
              <span
                className="MuiButton-label"
              >
                <span>
                  Prev
                </span>
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
              width="50"
            >
              <span
                className="MuiButton-label"
              >
                <span>
                  Next
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
    >
      <div
        aria-label="nav tabs example"
        className="MuiTabs-root"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": null,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-controls="nav-tabpanel-0"
              aria-disabled={false}
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
              href="/drafts"
              id="nav-tab-0"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper"
              >
                <span>
                  Ordered
                </span>
              </span>
              <span
                className="PrivateTabIndicator-root-227 PrivateTabIndicator-colorSecondary-229 MuiTabs-indicator"
                style={Object {}}
              />
            </a>
            <a
              aria-controls="nav-tabpanel-1"
              aria-disabled={false}
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
              href="/trash"
              id="nav-tab-1"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper"
              >
                <span>
                  Delivered
                </span>
              </span>
            </a>
            <a
              aria-controls="nav-tabpanel-2"
              aria-disabled={false}
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
              href="/spam"
              id="nav-tab-2"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
            >
              <span
                className="MuiTab-wrapper"
              >
                <span>
                  Cancelled
                </span>
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="nav-tab-0"
        className="MuiTypography-root MuiTypography-body1"
        hidden={false}
        id="nav-tabpanel-0"
        role="tabpanel"
      >
        <div
          className="MuiBox-root MuiBox-root-261"
        >
          <div
            className="makeStyles-tableWrapper-3"
          >
            <table
              aria-labelledby="tableTitle"
              className="MuiTable-root makeStyles-table-2"
            >
              <thead
                className="MuiTableHead-root"
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    aria-sort="ascending"
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Date
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Customer
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Address1
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Address2
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Country
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        State
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        City
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span>
                        Total
                      </span>
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root"
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-hover"
                  tabIndex={-1}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                    scope="row"
                  />
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    <a
                      href="/orderitems"
                      onClick={[Function]}
                    >
                      Thet Oo
                       
                      Maung
                    </a>
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    98 street
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Min Buu
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Myanmar
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Magway
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Magway
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    6000
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root MuiTableRow-hover"
                  tabIndex={-1}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                    scope="row"
                  />
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    <a
                      href="/orderitems"
                      onClick={[Function]}
                    >
                      Soe
                       
                      Thu
                    </a>
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Bauk taw
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    POL
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Myanmar
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Mandalay
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    Mandalay
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    2000
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="nav-tab-1"
        className="MuiTypography-root MuiTypography-body1"
        hidden={true}
        id="nav-tabpanel-1"
        role="tabpanel"
      >
        <div
          className="MuiBox-root MuiBox-root-286"
        />
      </div>
      <div
        aria-labelledby="nav-tab-2"
        className="MuiTypography-root MuiTypography-body1"
        hidden={true}
        id="nav-tabpanel-2"
        role="tabpanel"
      >
        <div
          className="MuiBox-root MuiBox-root-287"
        />
      </div>
    </div>
  </div>
</div>
`;
